kerl

```diff
@@ -2118,29 +2177,42 @@ github_download() {
     tarball_file="$KERL_DOWNLOAD_DIR/$2"
     tarball_url="$OTP_GITHUB_URL/archive/$2"
     prebuilt_url="$OTP_GITHUB_URL/releases/download/OTP-$1/otp_src_$1.tar.gz"
-    if curl --silent --location --fail -I "$prebuilt_url" >/dev/null; then
+    http_code=$(_curl /dev/null "$prebuilt_url" "-I")
+    if [ 200 = "$http_code" ]; then
         tarball_url=$prebuilt_url
         unset KERL_USE_AUTOCONF
     fi
     # if the file doesn't exist or the file has no size
     if [ ! -s "$tarball_file" ]; then
         notice "Downloading (from GitHub) Erlang/OTP $1 to $KERL_DOWNLOAD_DIR..."
-        curl -s -f -L -o "$tarball_file" "$tarball_url" || return 1
+        http_code=$(_curl "$tarball_file" "$tarball_url")
+        if [ 200 != "$http_code" ]; then
+            error "_curl $tarball_url returned $http_code!"
```

Comment:
same - i don't think we need to expose the internal function call name in our errors - they just need to say the http request returned X status or w/e

Reply:
8b1c6ab

Reply:
8b1c6ab